csrc/multidevice/utils.cpp

```diff
@@ -225,6 +225,30 @@ void shardAllLike(TensorView* ref, std::vector<TensorView*> tvs) {
 }
 } // namespace
 
+void propagateShardings(Fusion* fusion) {
+  for (auto expr : fusion->exprs()) {
+    auto inputs = ir_utils::filterByType<TensorView>(expr->inputs());
+    auto outputs = ir_utils::filterByType<TensorView>(expr->outputs());
+    std::vector<TensorView*> input_with_mesh;
```

Comment:
it is enough to simply allocate one pointer `TensorView*`, initialized to `nullptr`